### PR TITLE
RSDK-7077 fix bug in builtin nav service when moving to a sufficiently close waypoint

### PR DESCRIPTION
**Pull request by raybjork**
This PR fixes the bug where motion returns an error when we are close to the next goal waypoint and the builtin nav service enters an infinite loop of trying to navigate to it anyway.

### CODE REVIEW

File: services/motion/builtin/move_request.go [hunk 1]
```diff
@@ -718,13 +716,10 @@ func (ms *builtIn) createBaseMoveRequest(
...
 		if spatialmath.PoseAlmostCoincidentEps(goal.Pose(), startPose, motionCfg.planDeviationMM) {
-			return nil, errGoalWithinPlanDeviation
-		}
-	} else {
-		if spatialmath.OrientationAlmostEqual(goal.Pose().Orientation(), spatialmath.NewZeroPose().Orientation()) &&
-			spatialmath.PoseAlmostCoincidentEps(goal.Pose(), startPose, motionCfg.planDeviationMM) {
-			return nil, errGoalWithinPlanDeviation
+			return nil, motion.ErrGoalWithinPlanDeviation
 		}
+	} else if spatialmath.PoseAlmostEqualEps(goal.Pose(), startPose, motionCfg.planDeviationMM) {
```
Comment (nicksanford) [inline services/motion/builtin/move_request.go:721]: Do we not need `spatialmath.OrientationAlmostEqual(goal.Pose().Orientation(), spatialmath.NewZeroPose().Orientation())` to be true as well?

File: services/navigation/builtin/builtin.go [hunk 2]
```diff
@@ -529,7 +529,10 @@ func (svc *builtIn) moveToWaypoint(ctx context.Context, wp navigation.Waypoint,
 	cancelCtx, cancelFn := context.WithCancel(ctx)
 	defer cancelFn()
 	executionID, err := svc.motionService.MoveOnGlobe(cancelCtx, req)
-	if err != nil {
+	if errors.Is(err, motion.ErrGoalWithinPlanDeviation) {
+		// make an exception for the error that is raised when motion is not possible because already at goal.
+		return svc.waypointReached(cancelCtx)
```
Comment (nicksanford) [inline services/navigation/builtin/builtin.go:534]: Should we add a test case for this in the nav tests?

File: services/motion/builtin/move_request.go (hunk 1 above)
Comment (raybjork) [reply services/motion/builtin/move_request.go:721]: `PoseAlmostEqualEps` is the logical AND of these two functions, so just a small simplification here

File: services/navigation/builtin/builtin.go (hunk 2 above)
Comment (raybjork) [reply services/navigation/builtin/builtin.go:534]: Yes, I wasn't sure how to test for this but I can figure something out

Comment (raybjork) [reply services/navigation/builtin/builtin.go:534]: Added a test

**Comment by nicksanford** [comment]
@raybjork you have failing unit tests

File: services/navigation/builtin/builtin_test.go [hunk 3]
```diff
@@ -1371,6 +1371,38 @@ func TestStartWaypoint(t *testing.T) {
...
 
+	t.Run("motion error returned when within planDeviation results in visiting waypoint", func(t *testing.T) {
+		s := setupStartWaypoint(ctx, t, logger)
+		defer s.closeFunc()
+
+		s.injectMS.MoveOnGlobeFunc = func(ctx context.Context, req motion.MoveOnGlobeReq) (motion.ExecutionID, error) {
+			return uuid.Nil, motion.ErrGoalWithinPlanDeviation
+		}
+
+		cancelCtx, cancelFn := context.WithTimeout(ctx, time.Millisecond*10)
```
Comment (nicksanford) [inline services/navigation/builtin/builtin_test.go:1382]: I'd recommend 200 ms as sometimes the CI server can be over-utilized / under provisioned & I'd recommend adding more buffer to reduce the likelihood of flakey tests

**Review by nicksanford — APPROVED** [review]

**Pull request merged**